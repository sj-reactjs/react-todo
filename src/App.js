
import React from 'react'
import './App.css';
export const TodoApp = React.lazy(() => import('./components/TodoApp'))

function App() {
  console.log('render');
  return (
    <div className="App">
      <React.Suspense fallback={<>Loading...</>}>
        <TodoApp />
      </React.Suspense>
    </div>
  );
}

export default App;
