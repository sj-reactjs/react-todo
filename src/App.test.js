import { render, screen } from '@testing-library/react';
import App, { TodoApp } from './App';

describe('App component test', () => {
  it(`initaial render`, () => {
    render(<App />);
    const linkElement = screen.getByText(/loading.../i);
    expect(linkElement).toBeInTheDocument();
  });

  it(`suspense test`, async () => {
    const { asFragment } = render(<App />)

    const delayDom = await asFragment(<App />)
    expect(delayDom).toMatchSnapshot()

  })
})
