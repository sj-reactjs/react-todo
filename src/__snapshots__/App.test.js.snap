// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test suspense test 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <div
        class="todo-app theme-dark"
      >
        <header
          class="header"
        >
          <span
            class="logo"
          >
            <img
              alt="react-todo-app"
              data-todo-type="header-logo"
              src="/logo192.png"
            />
          </span>
          <span
            class="search-section"
          >
            <input
              placeholder="Search"
              type="text"
            />
            <i
              class="fa fa-search"
            />
            <button>
              Switch To 
              <i
                class="fa fa-sun-o"
              />
            </button>
          </span>
        </header>
        <div
          class="stats-card"
        >
          <span
            class="card"
          >
            <p>
              not-start
            </p>
            <p>
              2
            </p>
          </span>
          <span
            class="card"
          >
            <p>
              in-progess
            </p>
            <p>
              1
            </p>
          </span>
          <span
            class="card"
          >
            <p>
              complete
            </p>
            <p>
              1
            </p>
          </span>
        </div>
        <div>
          <form>
            <div>
              <div>
                <label
                  for="task"
                >
                  Task
                </label>
              </div>
              <input
                id="task"
                name="task"
                placeholder="Enter Task"
                type="text"
                value=""
              />
            </div>
            <div>
              <div>
                <label
                  for="description"
                >
                  Description
                </label>
              </div>
              <input
                id="description"
                name="description"
                placeholder="Description"
                value=""
              />
            </div>
            <div>
              <div>
                <label
                  for="status"
                >
                  Status
                </label>
              </div>
              <select
                id="status"
                name="status"
              >
                <option
                  disabled=""
                  value=""
                >
                  Select Status
                </option>
                <option
                  value="not-start"
                >
                  Not Start
                </option>
                <option
                  value="in-progress"
                >
                  In Progress
                </option>
                <option
                  value="complete"
                >
                  Complete
                </option>
              </select>
            </div>
            <div>
              <button
                type="submit"
              >
                Submit
              </button>
              <button
                type="button"
              >
                Cancel
              </button>
            </div>
          </form>
        </div>
        <section
          role="list"
        >
          <div
            style="display: flex;"
          >
            <div
              class="todo-card"
              role="listitem"
            >
              <div>
                <span
                  data-action="edit"
                  role="button"
                  tabindex="0"
                >
                  Edit
                </span>
                <span
                  class="created"
                >
                  N/A
                </span>
                <span
                  class="right close"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    x
                  </span>
                </span>
              </div>
              <div
                class="title"
              >
                <h3>
                  Lorem Ipsum 
                </h3>
              </div>
              <div
                class="description"
              >
                <p>
                  dolor sit amit
                </p>
              </div>
              <div
                class="status"
              >
                not-start
              </div>
              <div
                class="updated"
              >
                N/A
              </div>
            </div>
            <div
              class="todo-card"
              role="listitem"
            >
              <div>
                <span
                  data-action="edit"
                  role="button"
                  tabindex="0"
                >
                  Edit
                </span>
                <span
                  class="created"
                >
                  N/A
                </span>
                <span
                  class="right close"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    x
                  </span>
                </span>
              </div>
              <div
                class="title"
              >
                <h3>
                  Task 1
                </h3>
              </div>
              <div
                class="description"
              >
                <p>
                  Task 1- description
                </p>
              </div>
              <div
                class="status"
              >
                not-start
              </div>
              <div
                class="updated"
              >
                N/A
              </div>
            </div>
            <div
              class="todo-card"
              role="listitem"
            >
              <div>
                <span
                  data-action="edit"
                  role="button"
                  tabindex="0"
                >
                  Edit
                </span>
                <span
                  class="created"
                >
                  N/A
                </span>
                <span
                  class="right close"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    x
                  </span>
                </span>
              </div>
              <div
                class="title"
              >
                <h3>
                  Task 2
                </h3>
              </div>
              <div
                class="description"
              >
                <p>
                  Task 2- description
                </p>
              </div>
              <div
                class="status"
              >
                in-progess
              </div>
              <div
                class="updated"
              >
                N/A
              </div>
            </div>
            <div
              class="todo-card"
              role="listitem"
            >
              <div>
                <span
                  data-action="edit"
                  role="button"
                  tabindex="0"
                >
                  Edit
                </span>
                <span
                  class="created"
                >
                  N/A
                </span>
                <span
                  class="right close"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    x
                  </span>
                </span>
              </div>
              <div
                class="title"
              >
                <h3>
                  Task 3
                </h3>
              </div>
              <div
                class="description"
              >
                <p>
                  Task 3- description
                </p>
              </div>
              <div
                class="status"
              >
                complete
              </div>
              <div
                class="updated"
              >
                N/A
              </div>
            </div>
          </div>
        </section>
        <footer>
          Footer !!
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
